devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -298,6 +300,13 @@ public void run() {
         }
     }
 
+    private void ensureMavenVersion() throws MojoExecutionException {
+        String mavenVersion = session.getSystemProperties().getProperty("maven.version");
+        getLog().debug("Detected Maven Version: " + mavenVersion);
+        DefaultArtifactVersion detectedVersion = new DefaultArtifactVersion(mavenVersion);
+        mavenVersionEnforcer.enforce(getLog(), "3.5.3", detectedVersion);
```

Comment:
Wondering if we could find a way to push this from the POM instead of having it hard coded so that we are sure it's consistent with the version restriction we have in the build?
Maybe a filtered `.properties` file included in the Maven plugin artifact?

Reply:
We can extract it and read it from a property file. We already have a few properties like that (they have been moved apparently). Let me revisit this.

Reply:
Done! Can you have another look @gsmet